packages/contracts/src/contracts/current/protocol/Exchange/MixinSettlementProxy.sol

```diff
@@ -68,43 +77,39 @@ contract MixinSettlementProxy is
         )
     {
         makerTokenFilledAmount = getPartialAmount(takerTokenFilledAmount, order.takerTokenAmount, order.makerTokenAmount);
-        require(
-            TRANSFER_PROXY.transferFrom(
-                order.makerTokenAddress,
-                order.makerAddress,
-                takerAddress,
-                makerTokenFilledAmount
-            )
+        bytes32 orderHash = getOrderHash(order);
+
+        TRANSFER_PROXY.transferFrom(
+            order.makerAssetProxyData,
+            order.makerAddress,
+            takerAddress,
+            makerTokenFilledAmount
         );
-        require(
-            TRANSFER_PROXY.transferFrom(
-                order.takerTokenAddress,
-                takerAddress,
-                order.makerAddress,
-                takerTokenFilledAmount
-            )
+
+        TRANSFER_PROXY.transferFrom(
```

Comment:
Why were these requires removed? We can't assume every token transfer will revert on failure, so we should add back `bool` returns to all instances of `transferFrom` and keep the requires.

Reply:
The transfer method in `IAssetProxy` does not return a value, it either succeeds or throws. The implementation for ERC20 or the old TokenProxy does the return value checking.

I think it's cleaner this way because you can not forget to check the return value.